docs(data): update client docs (#1235)
regen-network/regen-ledger: x/data/spec/07_client.md

The `query` commands allow users to query `data` state.

For examples on how to query state using CLI, see the data module [Query commands](https://docs.regen.network/commands/regen_query_data.html) documentation.

### Transactions

The `tx` commands allow users to interact with the `data` module.

For examples on how to submit transactions using CLI, see the data module [Transaction commands](https://docs.regen.network/commands/regen_tx_data.html) documentation.

## gRPC

A user can query the `data` module using gRPC endpoints.

### AnchorByIRI

> The `AnchorByIRI` endpoint allows users to query anchored data by the content hash (i.e. IRI).

Author: ```suggestion
The `AnchorByIRI` endpoint allows users to query a data anchor by the IRI of the data.
```

```bash
regen.data.v1.Query/ConvertIRIToHash
```

Example:

```bash
grpcurl -plaintext -d '{"iri":"regen:13toVgf5aZqSVSeJQv562xkkeoe3rr3bJWa29PHVKVf77VAkVMcDvVd.rdf"}' 127.0.0.1:9090 regen.data.v1.Query/ConvertIRIToHash
```

Example Output:

```bash
{
  "contentHash": {
    "graph": {
      "hash": "YWJjZGVmZ2hpamtsbW5vcHFyc3R1dnd4eXoxMjM0NTY=",
      "digestAlgorithm": "DIGEST_ALGORITHM_BLAKE2B_256",
      "canonicalizationAlgorithm": "GRAPH_CANONICALIZATION_ALGORITHM_URDNA2015"
    }
  }
}
```

### ConvertHashToIRI

The `ConvertHashToIRI` endpoint allows users to convert ContentHash to an IRI.

```bash
regen.data.v1.ConvertHashToIRI
```

Example:

```bash
grpcurl -plaintext -d '{"content_hash": {"graph": {"hash": "YWJjZGVmZ2hpamtsbW5vcHFyc3R1dnd4eXoxMjM0NTY=","digest_algorithm": "DIGEST_ALGORITHM_BLAKE2B_256","canonicalization_algorithm": "GRAPH_CANONICALIZATION_ALGORITHM_URDNA2015"}}}' 127.0.0.1:9090 regen.data.v1.Query/ConvertHashToIRI
```

Example Output:

```bash
{
  "iri": "regen:13toVgf5aZqSVSeJQv562xkkeoe3rr3bJWa29PHVKVf77VAkVMcDvVd.rdf"
}
```

## REST

A user can query the `data` module using REST endpoints.

### anchor-by-iri

> The `anchor-by-iri` endpoint allows users to query anchored data by the content hash (i.e. IRI).

Author: ```suggestion
The `anchor-by-iri` endpoint allows users to query a data anchor by the IRI of the data.
```
